packages/dashboard/src/components/task-summary-panel.tsx

```diff
@@ -0,0 +1,252 @@
+      Object.keys(storedTasks).forEach((element) => {
+        if (!(element in currentContent)) {
+          taskContentsTemp[element] = storedTasks[element];
+        }
+      });
+      console.log(taskContentsTemp);
+      return taskContentsTemp;
+    });
+    savedTasksContent.current = {};
+  };
+
+  const determineStyle = (state: number): string => {
+    switch (state) {
+      case RomiCore.TaskSummary.STATE_QUEUED:
+        return classes.queued;
+      case RomiCore.TaskSummary.STATE_ACTIVE:
+        return classes.active;
+      case RomiCore.TaskSummary.STATE_COMPLETED:
+        return classes.completed;
+      case RomiCore.TaskSummary.STATE_FAILED:
+        return classes.failed;
+      default:
+        return 'UNKNOWN';
+    }
+  };
+
+  // Update Task list content
+  React.useEffect(() => {
+    const renderActor = (taskStatus: string): React.ReactElement | undefined => {
+      const actor = getActorFromStatus(taskStatus);
+      if (!actor) return;
+      return (
+        <Typography variant="body1" className={classes.taskActor}>
+          {actor}
+        </Typography>
+      );
+    };
+
+    const renderTaskTreeItem = (task: RomiCore.TaskSummary) => {
+      return (
+        <TreeItem
+          data-component="TreeItem"
+          nodeId={task.task_id}
+          key={task.task_id}
+          classes={{
+            label: `${determineStyle(task.state)} ${classes.labelContent}`,
+            root: classes.treeChildren,
+            expanded: classes.expanded,
+          }}
+          label={
+            <>
+              <Typography variant="body1" noWrap>
+                {task.state === RomiCore.TaskSummary.STATE_ACTIVE && (
+                  <IconButton
+                    // TODO: replace onClick with the pause plans logic.
+                    onClick={(e) => {
+                      e.preventDefault();
+                      console.log('Not implemented');
+                    }}
+                  >
+                    <PlayCircleFilledWhiteIcon />
+                  </IconButton>
+                )}
+                {task.state === RomiCore.TaskSummary.STATE_QUEUED && (
+                  <IconButton
+                    // TODO: replace onClick with the pause plans logic.
+                    onClick={(e) => {
+                      e.preventDefault();
+                      console.log('Not implemented');
+                    }}
+                  >
+                    <PauseCircleFilledIcon />
+                  </IconButton>
+                )}
+                {task.task_id}
+              </Typography>
+              {renderActor(task.status)}
+            </>
+          }
+        >
+          <TaskSummaryPanelItem task={task} key={task.task_id} />
+        </TreeItem>
+      );
+    };
+
+    if (allTasks.length === 0) return;
+
+    setTaskContents((currentContent) => {
+      // Assigning another reference.
+      let taskContentsTemp = Object.assign({}, currentContent);
```

Comment:
use `const`. Also let's rename this to `newTaskContents` to better reflect its purpose.